src/kvstore/RocksEngine.cpp

```diff
@@ -337,16 +330,16 @@ void RocksEngine::removePart(PartitionID partId) {
 
 std::vector<PartitionID> RocksEngine::allParts() {
     std::unique_ptr<KVIterator> iter;
-    static const size_t prefixLen = ::strlen(kSystemParts);
-    static const std::string prefixStr(kSystemParts, prefixLen);
+    static const std::string prefixStr = NebulaKeyUtils::systemPartsPrefix();
     CHECK_EQ(ResultCode::SUCCEEDED, this->prefix(prefixStr, &iter));
 
     std::vector<PartitionID> parts;
     while (iter->valid()) {
         auto key = iter->key();
-        CHECK_EQ(key.size(), prefixLen + sizeof(PartitionID));
-        parts.emplace_back(
-           *reinterpret_cast<const PartitionID*>(key.data() + key.size() - sizeof(PartitionID)));
+        CHECK_EQ(key.size(), sizeof(PartitionID));
+        auto partId = *reinterpret_cast<const PartitionID*>(key.data());
+        partId = partId >> 8;
```

Comment:
Please note that we should pay more attentions to do right shifting operations on a signed integer, since there are chances that the sign extention is not what you want. In my opinion, it's always a bad idea to make an integer like this signed.

Reply:
When we encoded the partition ID, using left shift firstly. 

Actually, partition ID is unsigned integer and it can't be a negative number so I think right shift maybe ok :)